CONTRIBUTING.md

```diff
@@ -550,6 +550,239 @@ cat generated/watcher/watcher.log  # See k_ids field for each core in the last d
+better to avoid that.
+
+You should first make sure main is up to date:
+
+```
+git checkout main
+git fetch origin
+git pull --rebase --prune
+```
+
+Then you can
+
+```
+git checkout user-123
+git rebase main
+```
+
+This will apply one commit at a time. Each commit is in order. If your branch
+has two commits, then the first one is applied, then the second one is applied.
+
+If there are no conflicts, everything will complete successfully. Then you can
+push the changes to origin. This is done through a forced push to save the
+rebase information:
+
+```
+git push -f
+```
+
+If there is any conflict with the commits being processed, you will need to
+edit the files to fix the problem. Information should be printed about what to
+do. It's probably a good idea not to skip commits.
+
+Don't be surprised if changes from a subsequent commit are not there in the
+first commit. For example, if you are editing the files to fix up the first of
+two commits, the files will not have the edits of the second commit. When
+editing files, only fix up the conflicts listed. Do not change anything else.
+
+If you do change anything else, then `git rebase --continue` will complain and
+you will probably have to restart.
+
+Look for HEAD. The conflict will look something like:
+
+```
+<<<< HEAD
+Some other edits
+====
+Your edits
+>>>> Your branch
+```
+
+Update the file to have a single piece of working code and remove the commit
+info. Make sure everything compiles and all the tests pass. Then you can
+continue with the rebase.
+
+```
+git rebase --continue
+```
+
+If something is wrong enough that you want to abort the rebase and undo all the
+changes, then you can start over. Do
+
+```
+git rebase --abort # go to before the rebase
+```
+
+If your local is in a bad state, you may also want to start from scratch on your
+local by pulling from origin, reflogging, or checking out a specific commit via
+its hash:
+
+```
+git pull --rebase # will undo changes
+git reflog
+git checkout <hash>
+```
+
+If none of those work you can also try:
+
+```
+git reset --hard origin/<BRANCH>
+```
+
+Note: If you are getting errors you may need to update the origin info via
+`git branch --set-upstream-to`.
+
+If everything goes well with all the updates. Then you can update origin:
+
+```
+git push -f
+```
+
+Note: It's okay to have a few commits, as long as each one works on its own.
+If you do want to combine commits you would want to run something like:
+
+```
+git rebase -i HEAD~10 # the number indicates how many commits you want to look at - here 10 commits
+```
+
+The latest one is at the bottom. You can use fixup or squash. Usually you want
+to use fixup (indicated by f) since that discards the message. Then edit the
+messages appropriately.
+
+However, new squash and merge functionality in the UI is much easier than
+doing this process manually. Therefore you should use the UI whenever possible.
+
+#### Merging to main
```

Comment:
Maybe I missed, want to make sure we have notes on PR description, workflow runs and linking workflow run and its status (via checkmark).

Reply:
These are in other sections of the document. E.g. Code reviews and Git rules and guidelines.